ultravox/data/datasets.py

```diff
@@ -805,6 +805,85 @@ def _get_sample(self, row) -> VoiceSample:
         return self._get_transcribe_sample(row, tcol="sentence")
 
 
+class CoVoST2Dataset(VoiceDataset):
+    """
+    CoVoST 2 is a large-scale multilingual speech translation corpus covering translations from 21 languages into English
+    and from English into 15 languages. The dataset is created using Mozilla's open-source Common Voice 4 database of
+    crowdsourced voice recordings. There are 2,900 hours of speech represented in the corpus.
+
+    The original Hugging Face dataset link: https://huggingface.co/datasets/facebook/covost2
+    Since this dataset requires audio files to be downloaded separately, a new dataset is created with the audio files:
+    https://huggingface.co/datasets/fixie-ai/covost2
+
+    Due to the scale of the dataset and the audio files being repeated, only a portion of the dataset was converted.
+    See [this issue](https://github.com/fixie-ai/ultravox/issues/50) for more information.
+
+    Supported subsets (En -> X):
+        'en_de', 'en_tr', 'en_fa', 'en_sv-SE', 'en_mn', 'en_zh-CN', 'en_cy',
+        'en_ca', 'en_sl', 'en_et', 'en_id', 'en_ar', 'en_ta', 'en_lv', 'en_ja'
+    Supported subsets (X -> En):
+        'fr_en', 'zh-CN_en', 'es_en'
+    """
+
+    CODE_TO_LANG = {
+        "en": "English",
+        "de": "German",
+        "tr": "Turkish",
+        "fa": "Persian",
+        "sv-SE": "Swedish",
+        "mn": "Mongolian",
+        "zh-CN": "Chinese",
+        "cy": "Welsh",
+        "ca": "Catalan",
+        "sl": "Slovenian",
+        "et": "Estonian",
+        "id": "Indonesian",
+        "ar": "Arabic",
+        "ta": "Tamil",
+        "lv": "Latvian",
+        "ja": "Japanese",
+        "fr": "French",
+        "es": "Spanish",
+    }
+
+    # We currently don't use this dataset for training, so mainly the first prompt it ever used.
```

Comment:
I didn't understand this comment.

Reply:
I was just noting that since we only use CoVoST for validation/testing and since there we always pin `num_prompts` to 1, that means we never use the other prompts. I still kept them just in case though.

Reply:
Agree that we would only use CoVoST for validation/testing for now, but eventually we may also want to use it for training, similar to what's reported in the llama3 paper. 

by the way, some language pairs already have the training split included.

Reply:
Yeah we can always add them if need be. The code is attached in the issue and it's only a few lines of code.

Before doing that I wanted to explore and see if we can somehow reduce the amount of duplicated data, which at the current rate was a bit ridiculous (e.g. using a custom dataset loader). Large datasets are quite hard to manage as any small modification can require downloading and uploading huge amounts of data.

As for the included `train` sets, I kept the `XX to EN` sets since there's no duplication anyway.
For the `EN to XX` I kept only one (or two maybe?). All the other audio samples are duplicated.